Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/get.go

```diff
@@ -323,3 +355,32 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	}
 	return utilerrors.NewAggregate(allErrs)
 }
+
+// mustPrintWithKinds determines if printer output
+// should print multiple kinds of objects, even if
+// "all-kinds" flag is not set
+func mustPrintWithKinds(objs []runtime.Object, infos []*resource.Info, sorter *kubectl.RuntimeSort) bool {
+	var lastMap *meta.RESTMapping
+	var withKind bool
+
+	if withKind {
+		return withKind
+	}
+
+	for ix := range objs {
+		var mapping *meta.RESTMapping
+		if sorter != nil {
+			mapping = infos[sorter.OriginalPosition(ix)].Mapping
+		} else {
+			mapping = infos[ix].Mapping
+		}
```

Comment:
I feel this is some not very trivial logic. A similar loop was part of `RunGet` before, so it probably works and very likely has no tests.

Now that we have this `mustPrintWithKinds` function, it wouldn't be hard to write a few unit tests that exercise it, so that we also make sure this logic here is reasonable.

We could have a few objects all of the same kind, and `mustPrintWithKinds` should return false, and another test case where we have two or more different kinds, and `mustPrintWithKinds` returns true. And another test where sorter != nil so we exercise the two branches of the if.